apps-rendering/src/adSlot.tsx

```diff
@@ -0,0 +1,131 @@
+import type { SerializedStyles } from '@emotion/react';
+import { css, ThemeProvider } from '@emotion/react';
+import {
+	background,
+	text,
+} from '@guardian/common-rendering/src/editorialPalette';
+import type { ArticleFormat } from '@guardian/libs';
+import {
+	from,
+	headline,
+	remSpace,
+	textSans,
+	until,
+} from '@guardian/source-foundations';
+import { Button, buttonThemeBrandAlt } from '@guardian/source-react-components';
+import type { FC, ReactElement } from 'react';
+import { darkModeCss, wideContentWidth } from 'styles';
+
+type Props = {
+	className: string;
+	paragraph: number;
+	format: ArticleFormat;
+};
+
+const adLabelsStyles = (format: ArticleFormat): SerializedStyles => css`
+	${textSans.xsmall()}
+	color: ${text.adLabel(format)};
+	padding: ${remSpace[3]};
+	float: left;
+	width: calc(100% - ${remSpace[6]});
```

Comment:
I don’t know if there’s a contract about spaces growing linearly. If what you’re trying to capture is that there’s padding on both sides, and you’re already using `calc`, it would be safer to be explicit and let the css calculate twice whatever `remSpace[3]` is.

```suggestion
	// We need to account for padding on both sides.
	width: calc(100% - 2 * ${remSpace[3]});
```

Reply:
Yes, that’s right, I knew that not all values were available–there’s no `space[8]`–so just wanted to make sure. Either is fine in this case!

```ts
export const space = {
	1: 4,
	2: 8,
	3: 12,
	4: 16,
	5: 20,
	6: 24,
	9: 36,
	12: 48,
	24: 96,
} as const;
```